contracts/src/management/ManagementContract.sol

```diff
@@ -15,125 +15,39 @@ contract ManagementContract {
     bool private isWithdrawalAvailable;
 
+    uint256 public lastBatchSeqNo = 0;
+    bytes32 private headRollupHash;
+
+    Structs.RollupStorage private rollups;
     //The messageBus where messages can be sent to Obscuro
     MessageBus.IMessageBus public messageBus;
     constructor() {
         messageBus = new MessageBus.MessageBus();
         emit LogManagementContractCreated(address(messageBus));
     }
 
-    // InitializeTree starts the list and sets the initial values
-    function InitializeTree(Structs.MetaRollup memory r) public {
-        require(!tree.initialized, "cannot be initialized again");
-        tree.initialized = true;
-
-        // TreeElement starts a 1 and has no parent ( ParentID: 0 )
-        tree.rollups[1] = Structs.TreeElement(1, 0, r);
-        tree._HEAD = 1;
-        tree._nextID = 2;
-        tree.rollupsHashes[r.Hash] = 1;
-
-        // withdrawals are available at the start
-        isWithdrawalAvailable = true;
+    function GetRollupByHash(bytes32 rollupHash) view public returns(bool, Structs.MetaRollup memory) {
+        Structs.MetaRollup memory rol = rollups.byHash[rollupHash];
+        return (rol.Hash == rollupHash , rol);
     }
 
-    function GetRollupByID(uint256 rollupID) view public returns(bool, Structs.TreeElement memory) {
-        Structs.TreeElement memory rol = tree.rollups[rollupID];
-        return (rol.ElementID != 0 , rol);
-    }
-
-    function GetRollupByHash(bytes32 rollupHash) view public returns (bool, Structs.TreeElement memory) {
-        return GetRollupByID(tree.rollupsHashes[rollupHash]);
-    }
-
-    function GetHeadRollup() internal view returns ( Structs.TreeElement memory ) {
-        return tree.rollups[tree._HEAD];
-    }
-
-    function GetParentRollup(Structs.TreeElement memory element) view public returns( bool, Structs.TreeElement memory) {
-        return GetRollupByID(element.ParentID);
-    }
-
-    function AppendRollup(uint256 _parentID, Structs.MetaRollup calldata _r) internal {
-        // guarantee the storage ids are not compromised
-        uint rollupID = tree._nextID;
-        tree._nextID++;
-
-        // cannot append to non-existing parent rollups
-        (bool found, Structs.TreeElement memory parent) = GetRollupByID(_parentID);
-        require(found, "parent not found");
-
-        // store the rollup in an element
-        tree.rollups[rollupID] = Structs.TreeElement({ 
-            ElementID: rollupID, 
-            ParentID: _parentID,
-            rollup: _r
-        });
-
-        // mark the element as a child of parent
-        tree.rollupChildren[_parentID].push(rollupID);
-
-        // store the hashpointer
-        tree.rollupsHashes[_r.Hash] = rollupID;
-
-        // mark this as the head
-        if (parent.ElementID == tree._HEAD) {
-            tree._HEAD = rollupID;
-        }
+    function GetHeadRollup() internal view returns ( Structs.MetaRollup memory ) {
```

Comment:
why is there a "head rollup" if there is no chain?

Reply:
I'm not sure if this is needed at this point. Technically there is a "sequence" of rollups with ever increasing batch sequence numbers so there is a head rollup. I will remove it given that its internal and not really used anywhere.